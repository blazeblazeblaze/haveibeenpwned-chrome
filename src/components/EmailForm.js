import React, { Component } from 'react';
import PropTypes from 'prop-types';

class EmailForm extends Component {
  constructor(props) {
    super(props);

    this.state = {
      email: ''
    };

    this.handleChange = this.handleChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
  }

  handleChange(event) {
    this.setState({email: event.target.value});
  }

  handleSubmit(event) {
    event.preventDefault();
    const { email } = this.state;
    if (email.length === 0) return;
    this.props.onSubmit(email)
  }

  render() {
    const {email} = this.state;

    return (
      <form onSubmit={this.handleSubmit} className="pure-form">
        <input type="email" className="pure-input-rounded" placeholder="Email" value={email} onChange={this.handleChange}/>
        {' '}
        <button type="submit" className="pure-button pure-button-primary">Pwned?</button>
      </form>
    );
  }
}

EmailForm.propTypes = {
  onSubmit: PropTypes.func.isRequired
};

export default EmailForm;
