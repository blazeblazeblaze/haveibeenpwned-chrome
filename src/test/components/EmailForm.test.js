import React from 'react';
import ReactDOM from 'react-dom';
import EmailForm from '../../components/EmailForm';
import renderer from 'react-test-renderer';

test('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<EmailForm onSubmit={jest.fn} />, div);
  ReactDOM.unmountComponentAtNode(div);
});

test('EmailForm renders the form', () => {
  const component = renderer.create(
    <EmailForm onSubmit={jest.fn} />
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
