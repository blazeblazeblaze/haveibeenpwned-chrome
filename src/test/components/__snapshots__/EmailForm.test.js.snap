// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmailForm renders the form 1`] = `
<form
  className="pure-form"
  onSubmit={[Function]}
>
  <input
    className="pure-input-rounded"
    onChange={[Function]}
    placeholder="Email"
    type="email"
    value=""
  />
   
  <button
    className="pure-button pure-button-primary"
    type="submit"
  >
    Pwned?
  </button>
</form>
`;
